app/services/hyrax/work_resource_query_service.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+module Hyrax
+  # Responsible for retrieving information based on the given work.
+  #
+  # @see ProxyDepositRequest
+  # @see SolrDocument
+  # @see Hyrax::SolrService
+  # @note This was extracted from the ProxyDepositRequest, which was coordinating lots of effort. It was also an ActiveRecord object that required lots of Fedora/Solr interactions.
+  class WorkResourceQueryService
+    # @param [String] id - The id of the work
+    def initialize(id:)
+      @id = id
+    end
+    attr_reader :id
+
+    # @return [Boolean] if the work has been deleted
+    def deleted_work?
+      Hyrax.query_service.find_by(id: id)
+      false
+    rescue Valkyrie::Persistence::ObjectNotFoundError
+      true
+    end
+
+    def work
+      # Need to ensure it is a work?
+      resource = Hyrax.query_service.find_by(id: id)
+      unless Hyrax.config.curation_concerns.include?(resource.class) ||
+             (resource.class.respond_to?(:_canonical_valkyrie_model) && Hyrax.config.curation_concerns.include?(resource.class._canonical_valkyrie_model)) # Wings-wrapped models
```

Comment:
could `#name` be used instead? the ORMConverter base passes this through to `#_canonical_valkyrie_model` when appropriate. i think it would be something like ` unless Hyrax.config.curation_concerns.include?(resource.class.name)`

Reply:
I had to change this to `#to_s` due to some failing tests where the Wings wrapping class returned `Hyrax::Work` for `#name` but `#to_s` returned `GenericWork`.

Reply:
Ugh, more failing tests so might have to check both `#name` and `#to_s`.